### ISSUE TITLE
Gitlab scopes not working

### ISSUE THREAD

**Issue by jdecool** [laravel/socialite#511]
- Socialite Version: 5.1.2
- Laravel Version: 8.20.0
- PHP Version: 8.0.0
- Database Driver & Version: MySQL 8.0

I reopen issue #507 because I confirm the bug.

I push minimal code to reproduce the bug in [this repository](https://github.com/jdecool/socialite-gitlab-bug/blob/main/routes/web.php#L21-L30).
I also push [my credentials for the sample app](https://github.com/jdecool/socialite-gitlab-bug/blob/main/.env.example#L7-L8).

### Description:

Defining scopes with Gitlab provider not working because of an incorrect URL parameter format.

The Gitlab scope separator should be `+` and not `,` like it is by default.

But there is a second problem. The `+` character is transformed into `%2B` when the URL is created by `http_build_query`.

### Steps To Reproduce:

Specify two scopes in the redirect action:

```php
return Socialite::driver('gitlab')
    ->setScopes(['read_user', 'read_api'])
    ->redirect();
```

Result this error:

![image](https://user-images.githubusercontent.com/433926/103045197-d4e8f000-4583-11eb-9357-170bc2e6694d.png)

Because of this parameter `&scope=read_user%2Bread_api` in the URL.
Should be `&scope=read_user+read_api`.

**Comment by driesvints** [comment]
I can't even find the scopes in the GitLab documentation. Can you point me to their docs where they show an example of using scopes and the available scopes?

Also: can you recreate that repo but commit your custom changes from the skeleton separately? Otherwise it's hard to see what you've changed.

**Comment by jdecool** [comment]
> I can't even find the scopes in the GitLab documentation. 

Scopes are not listed in the documentation.
They are listed when you want to create a new application through this address: https://gitlab.com/-/profile/applications

![image](https://user-images.githubusercontent.com/433926/103284489-a764d200-49db-11eb-8845-9c31cbace151.png)

Documentation for the authorization workflow is available a this address => https://docs.gitlab.com/ee/api/oauth2.html

Extract > _This will ask the user to approve the applications access to their account based on the scopes specified in REQUESTED_SCOPES and then redirect back to the REDIRECT_URI you provided. The scope parameter is a space separated list of scopes you want to have access to (e.g. scope=read_user+profile would request read_user and profile scopes)_. 

> can you recreate that repo but commit your custom changes from the skeleton separately?

Done. You can check [this commit](https://github.com/jdecool/socialite-gitlab-bug/commit/640eaf013426eb37b53bb04b9f732ff037dd2fcc).
Credentials in the commit is a dedicated test app :)

**Comment by jdecool** [comment]
I close this issue as it should be fixed by #512
